plugins/drivers/plugin.go

```diff
@@ -19,7 +21,7 @@ type PluginDriver struct {
 	logger hclog.Logger
 }
 
-func NewDriverPlugin(d DriverPlugin, logger hclog.Logger) plugin.GRPCPlugin {
```

Comment:
Wasn't obvious to me why this returns an interface and felt unnecessary when we use concrete `PluginDriver` type and always cast it down on this call.